libs/entitlement/src/main/java/org/elasticsearch/entitlement/runtime/policy/FileUtils.java

```diff
@@ -9,17 +9,37 @@
 
 package org.elasticsearch.entitlement.runtime.policy;
 
+import org.elasticsearch.core.Strings;
 import org.elasticsearch.core.SuppressForbidden;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
 
 import java.io.File;
 import java.util.Comparator;
+import java.util.function.BiPredicate;
 
 import static java.lang.Character.isLetter;
 
 public class FileUtils {
 
+    private static final Logger logger = LogManager.getLogger(FileUtils.class);
+
     private FileUtils() {}
 
+    interface CharComparator {
+        int compare(char c1, char c2);
+    }
+
+    private static final CharComparator CHARACTER_COMPARATOR = Platform.WINDOWS.isCurrent()
```

Comment:
Using static finals here should make code perform a little bit better as the JIT as a better chance of optimizing this. But the downside is that this is harder to test: you need to run tests on a Windows or Linux machine to test case sensitive/insensitive.
@rjernst what's your take on this?
Would you prefer to have a non-static comparator? We could still instantiate it as final in FileAccessTree, and create the correct final predicates in the ctor. Or have 2 different subclasses, and create the correct instance (CaseInsensitveComparator vs CaseSensitiveComparator)